float64[2] joint1
float64[2] joint2
float64[2] joint3
bool[2] legal
